src/harness/sas.py

```diff
@@ -304,9 +304,124 @@ def InjectCpiUser(self, request):
     _RequestPost('https://%s/admin/injectdata/cpi_user' % self._base_url,
                  request, self._tls_config)
 
+  def TriggerCreateActivityDump(self, request):
+    _RequestPost('https://%s/admin/trigger/create_full_activity_dump' % self._base_url,
+                 request, self._tls_config)
+
+  def InjectPeerSas(self,request):
+    _RequestPost('https://%s/admin/injectdata/peer_sas' % self._base_url,
+                 request, self._tls_config)
+
   def _GetDefaultAdminSSLCertPath(self):
     return os.path.join('certs', 'admin_client.cert')
 
   def _GetDefaultAdminSSLKeyPath(self):
     return os.path.join('certs', 'admin_client.key')
 
+def GetServer():
+  config_parser = ConfigParser.RawConfigParser()
+  config_parser.read(['sas.cfg'])
+  server_base_url = config_parser.get('HttpServer', 'BaseUrl')
+  server_port = int(config_parser.get('HttpServer', 'Port'))
+  return HttpServer({'baseUrl': server_base_url, 'port': server_port})
+
+class HttpServer(sas_interface.HttpServerInterface):
+  """Test Harness acting as a Http Server to receive Pull/GET and Push/POST
+   Requests from SAS Under Test
+  """
+  def __init__(self, server_details):
+    self._server_details = server_details
+    self._request_event = threading.Event()
+    self.parameters = {}
+    self.response = None
+
+  def _callbackHandler(self, response=None):
+    """Release wait initiated by _requestResponse when there is GET/POST Request from
+    SAS Under Test"""
+    self.response = response
+    self._request_event.set()
+
+  def _handleRequest(self, callbackHandler, getSetupParameters):
+    return lambda *args: HttpServerHandler(callbackHandler, getSetupParameters, *args)
+
+  def _getSetupParameters(self):
+    return self.parameters
+
+  def _requestResponse(self):
+    """Initiate wait till timeout to get the response from SAS Under Test and then
+    return the response as soon as _callbackHandler is called"""
+    logging.info("Waiting for the Response from SAS Under Test")
+    self._request_event.wait(20.0)
+    return self.response
+
+  def setupServer(self, parameters=None):
+    self.parameters = parameters
+    return self._requestResponse
+
+  def getBaseUrl(self):
+    return 'https://%s:%d' % (self._server_details['baseUrl'], self._server_details['port'])
+
+  def StartServer(self):
+    request_handler = self._handleRequest(self._callbackHandler, self._getSetupParameters)
+    self.server = HTTPServer((self._server_details['baseUrl'], self._server_details['port']),request_handler)
+    cert_file = 'server.cert'
```

Comment:
This will also need to be configurable as we may have multiple SAS Test Harnesses, each with its own cert.

Reply:
Already make it configurable for the input of startServer function in util.py. People can give this config in extra.config file for the test case